cassovary-collections/src/main/scala/com/twitter/cassovary/collections/SortedCSeqOps.scala

```diff
@@ -80,13 +96,8 @@ object SortedArrayOps {
       }
     }
 
-    slice(union, 0, resultIdx)
-  }
-
-  @inline def slice(a: Array[Int], from: Int, len: Int): Array[Int] = {
-    val result = Array.ofDim[Int](len)
-    Array.copy(a, from, result, 0, len)
-    result
+    // possible memory leak
```

Comment:
probably copy to new array (also add a comment about this tradeoff / memory leak)